source/level_set_okz_compute_normal.cc

```diff
@@ -173,12 +173,16 @@ LevelSetOKZSolverComputeNormal<dim>::compute_normal_vmult(
 #undef OPERATION
 
   // The number "4" below is so that constraints_normals is used
-  for (unsigned int i = 0; i < this->matrix_free.get_constrained_dofs(4).size(); ++i)
+  for (unsigned int i = 0;
+       i < this->matrix_free.get_constrained_dofs(parameters.dof_index_normal).size();
+       ++i)
     for (unsigned int d = 0; d < dim; ++d)
-      dst.block(d).local_element(this->matrix_free.get_constrained_dofs(4)[i]) =
+      dst.block(d).local_element(
+        this->matrix_free.get_constrained_dofs(parameters.dof_index_normal)[i]) =
         preconditioner.get_vector().local_element(
-          this->matrix_free.get_constrained_dofs(4)[i]) *
-        src.block(d).local_element(this->matrix_free.get_constrained_dofs(4)[i]);
+          this->matrix_free.get_constrained_dofs(parameters.dof_index_normal)[i]) *
+        src.block(d).local_element(
+          this->matrix_free.get_constrained_dofs(parameters.dof_index_normal)[i]);
```

Comment:
Can we make the code a bit more compact like this:
```c++
  for (const unsigned int entry : this->matrix_free.get_constrained_dofs(parameters.dof_index_normal))
    dst.block(d).local_element(entry) = preconditioner.get_vector().local_element(entry) *
                                        src.block(d).local_element(entry);
```

Reply:
Of course. I cleaned up the corresponding places. In addition, I've run the MeltPoolDG test suite and it seems to be fine.